src/aihwkit/utils/analog_info.py

```diff
@@ -0,0 +1,287 @@
+        if list(module.children()) == []:
+            module.register_forward_hook(hook)
+        else:
+            for _, layer in module.named_children():
+                self.register_hooks_recursively(layer, hook)
+
+    def create_layer_summary(self) -> List[LayerInfo]:
+        """Create the layer summary list.
+
+        This list contains LayerInfo elements that corresponds to each layer of the model."""
+        layer_summary = []
+
+        def get_size_hook(_: nn.Module, _input: INPUT_SIZE_TYPE, _output: INPUT_SIZE_TYPE) -> None:
+            nonlocal layer_summary
+            input_size = list(_input[0].size())
+            output_size = list(_output.size())
+            layer_summary.append(LayerInfo(_, self.rpu_config, input_size, output_size))
+
+        self.register_hooks_recursively(self.model, get_size_hook)
+        dummy_var = torch.zeros(self.input_size)
+        self.model(dummy_var)
+        return layer_summary
+
+    def calculate_num_tiles(self) -> int:
+        """Calculate the total number of tiles needed by the model."""
+        total_tile_number = 0
+        for x in self.layer_summary:
+            total_tile_number += x.num_tiles
+        return total_tile_number
+
+    def calculate_num_analog(self) -> int:
+        """Calculate the number of analog layers."""
+        total_nb_analog = 0
+        for x in self.layer_summary:
+            total_nb_analog += 1 if x.isanalog else 0
+        return total_nb_analog
+
+    def __repr__(self) -> str:
+        """Print summary results."""
+        divider = "=" * FORMATTING_WIDTH + "\n"
+        name = "Model Name: " + self.model.__class__.__name__ + "\n"
+        result = divider + name + divider
+        result += "Per-layer Information\n" + divider
+
+        # Add header
+        header = [*COLUMN_NAMES.values()]
+        for i, category in enumerate(COLUMN_DEFINITIONS):
+            header_i = [v for x, v in header if x == i]
+            trim_length = (COLUMN_WIDTH*len(header_i) - len(category))
+            result += category + " "*trim_length
+            if i == len(COLUMN_DEFINITIONS)-1:
+                break
+            result += '| '
+        result += "\n"+divider
+        for i, category in enumerate(COLUMN_DEFINITIONS):
+            header_i = [v for x, v in header if x == i]
+            result += (("{:<20}"*len(header_i)).format(*header_i))
+        result += "\n"
+
+        for x in self.layer_summary:
+            result += repr(x)
+
+        result += divider
+        result += "General Information\n" + divider
+        result += "Total number of tiles: " + str(self.total_tile_number) + "\n"
+        result += "Total number of analog layers: " + str(self.total_nb_analog) + "\n"
+        return result
+
+
+def analog_summary(model: nn.Module,
+                   input_size: Optional[INPUT_SIZE_TYPE] = None,
+                   rpu_config: Optional[RPUConfigAlias] = None) -> AnalogInfo:
+    """
```

Comment:
Is there an easy way for the use to select the columns displayed? like torchinfos "col_names". https://pypi.org/project/torchinfo/
But we can also tackle this as a follow-up